azkaban-common/src/main/java/azkaban/executor/selector/ExecutorComparator.java

```diff
@@ -0,0 +1,307 @@
+    comparatorCreatorRepository.put(PRIORITY_COMPARATOR_NAME, new ComparatorCreator(){
+      @Override public FactorComparator<Executor> create(int weight) { return getRemainingTmpSizeComparator(weight); }});
+
+    // register the creator for last dispatched time comparator.
+    comparatorCreatorRepository.put(LSTDISPATCHED_COMPARATOR_NAME, new ComparatorCreator(){
+      @Override public FactorComparator<Executor> create(int weight) { return getLstDispatchedTimeComparator(weight); }});
+
+    // register the creator for CPU Usage comparator.
+    comparatorCreatorRepository.put(CPUUSAGE_COMPARATOR_NAME, new ComparatorCreator(){
+      @Override public FactorComparator<Executor> create(int weight) { return getCpuUsageComparator(weight); }});
+  }
+
+
+  /**
+   * constructor of the ExecutorComparator.
+   * @param comparatorList   the list of comparator, plus its weight information to be registered,
+   *  the parameter must be a not-empty and valid list object.
+   * */
+  public ExecutorComparator(Map<String,Integer> comparatorList) {
+    if (null == comparatorList|| comparatorList.size() == 0){
+      logger.error("failed to initialize executor comparator as the passed comparator list is invalid or empty.");
+      throw new IllegalArgumentException("filterList");
+    }
+
+    // register the comparators, we will now throw here if the weight is invalid, it is handled in the super.
+    for (Entry<String,Integer> entry : comparatorList.entrySet()){
+      if (comparatorCreatorRepository.containsKey(entry.getKey())){
+        this.registerFactorComparator(comparatorCreatorRepository.
+            get(entry.getKey()).
+            create(entry.getValue()));
+      } else {
+        logger.error(String.format("failed to initialize executor comparator as the comparator implementation for requested factor '%s' doesn't exist.",
+            entry.getKey()));
+        throw new IllegalArgumentException("comparatorList");
+      }
+    }
+  }
+
+  @Override
+  public String getName() {
+    return "ExecutorComparator";
+  }
+
+  private interface ComparatorCreator{
+    FactorComparator<Executor> create(int weight);
+  }
+
+  /**
+   * helper function that does the object  on two statistics, comparator can leverage this function to provide
+   * shortcuts if   the statistics object is missing from one or both sides of the executors.
+   * @param stat1   the first statistics  object to be checked .
+   * @param stat2   the second statistics object to be checked.
+   * @param caller  the name of the calling function, for logging purpose.
+   * @param result  result Integer to pass out the result in case the statistics are not both valid.
+   * @return true if the passed statistics are NOT both valid, a shortcut can be made (caller can consume the result),
+   *         false otherwise.
+   * */
+  private static boolean statisticsObjectCheck(Statistics statisticsObj1, Statistics statisticsObj2, String caller, Integer result){
```

Comment:
I think the comment is not up to date?

also given that the return type is boolean, I am assuming the method name would something like isXXX, or hasYYY